nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmUploadHandler.java

```diff
@@ -86,21 +84,9 @@ public UploadResponse handle(Repository repository, ComponentUpload upload) thro
     String fileName = payload.getName() != null ? payload.getName() : StringUtils.EMPTY;
 
     try (TempBlob tempBlob = storageFacet.createTempBlob(payload, HASH_ALGORITHMS)) {
-      HelmAttributes attributesFromInputStream;
-      AssetKind assetKind;
-      String extension;
-
-      if (fileName.endsWith(PROVENANCE_EXTENSION)) {
-        attributesFromInputStream = helmPackageParser.getAttributesProvenanceFromInputStream(tempBlob.get());
-        assetKind = AssetKind.HELM_PROVENANCE;
-        extension = PROVENANCE_EXTENSION;
-      } else if (fileName.endsWith(TGZ_EXTENSION)) {
-        attributesFromInputStream = helmPackageParser.getAttributesFromInputStream(tempBlob.get());
-        assetKind = AssetKind.HELM_PACKAGE;
-        extension = TGZ_EXTENSION;
-      } else {
-        throw new IllegalArgumentException("Unsupported extension. Extension must be .tgz or .tgz.prov");
-      }
+      AssetKind assetKind = AssetKind.getAssetKindByFileName(fileName);
+      HelmAttributes attributesFromInputStream = helmPackageParser.getAttributes(assetKind, tempBlob.get());
+      String extension = assetKind.getExtension();
 
```

Comment:
You forgot validation for an extension.  
Unsupported extension. Extension must be .tgz or .tgz.prov

Reply:
Done